File: 57_Literature_Quiz/perl/litquiz.pl
#!/usr/bin/perl
use strict;


print ' 'x 25 . "LITERATURE QUIZ\n";
print ' 'x 15 . "CREATIVE COMPUTING MORRISTOWN, NEW JERSEY\n";
print "\n"; print "\n"; print "\n";
print "TEST YOUR KNOWLEDGE OF CHILDREN'S LITERATURE.\n";
print "\n"; print "THIS IS A MULTIPLE-CHOICE QUIZ.\n";
print "TYPE A 1, 2, 3, OR 4 AFTER THE QUESTION MARK.\n";
print "\n"; print "GOOD LUCK!\n";
my $R=0;


print "\n"; print "\n";
print "IN PINOCCHIO, WHAT WAS THE NAME OF THE CAT\n";
print "1)TIGGER, 2)CICERO, 3)FIGARO, 4)GUIPETTO";
print "? "; chomp(my $A = <STDIN>);

if ($A eq 3) {
	$R++;
	print "VERY GOOD! HERE'S ANOTHER.\n";
	} else {
	print "SORRY...FIGARO WAS HIS NAME.\n";
	}


print "\n"; print "\n";
print "FROM WHOSE GARDEN DID BUGS BUNNY STEAL THE CARROTS?\n";
print "1)MR. NIXON'S, 2)ELMER FUDD'S, 3)CLEM JUDD'S, 4)STROMBOLI'S";
print "? "; chomp($A = <STDIN>);

if ($A eq 2) {
	print "PRETTY GOOD!\n";
	$R=$R+1;
	} else {
	print "TOO BAD...IT WAS ELMER FUDD'S GARDEN.\n";
	}


print "\n"; print "\n";
print "IN THE WIZARD OF OS, DOROTHY'S DOG WAS NAMED\n";
print "1)CICERO, 2)TRIXIA, 3)KING, 4)TOTO";
print "? "; chomp($A = <STDIN>);
if ($A eq 4) {
	print "YEA! YOU'RE A REAL LITERATURE GIANT.\n";
	$R=$R+1;
	} else {
	print "BACK TO THE BOOKS,...TOTO WAS HIS NAME.\n";
	}


print "\n"; print "\n";
print "WHO WAS THE FAIR MAIDEN WHO ATE THE POISON APPLE\n";
print "1)SLEEPING BEAUTY, 2)CINDERELLA, 3)SNOW WHITE, 4)WENDY";
print "? "; chomp($A = <STDIN>);
if ($A eq 3) {
	print "GOOD MEMORY!\n";
	$R=$R+1;
	} else {
	print "OH, COME ON NOW...IT WAS SNOW WHITE.\n";
	}


print "\n"; print "\n";
if ($R eq 4) {
	print "WOW! THAT'S SUPER! YOU REALLY KNOW YOUR NURSERY\n";
	print "YOUR NEXT QUIZ WILL BE ON 2ND CENTURY CHINESE\n";
	print "LITERATURE (HA, HA, HA)\n";
	exit
	}
if ($R<2) {
	print "UGH. THAT WAS DEFINITELY NOT TOO SWIFT. BACK TO\n";
	print "NURSERY SCHOOL FOR YOU, MY FRIEND.\n";
	exit;
	}

print "NOT BAD, BUT YOU MIGHT SPEND A LITTLE MORE TIME\n";
print "READING THE NURSERY GREATS.\n";
exit;

File: 57_Literature_Quiz/javascript/litquiz.mjs
#!/usr/bin/env node

import { println, input } from '../../00_Common/javascript/common.mjs';

function printAlign(message = "", align = "left") {
    // process.stdout.columns is the number of spaces per line in the terminal
    const maxWidth = process.stdout.columns
    if (align === "center") {
        // calculate the amount of spaces required to center the message
        const padColCount = Math.round((process.stdout.columns-message.length)/2);
        const padding = padColCount <= 0 ? '' : ' '.repeat(padColCount);
        println(padding, message);
    } else if (align === "right") {
        const padColCount = Math.round(process.stdout.columns-message.length);
        const padding = padColCount <= 0 ? '' : ' '.repeat(padColCount);
        println(padding, message);
    } else {
        println(message);
    }
}

function equalIgnoreCase(correct, provided){
    return correct.toString().toLowerCase() === provided.toString().toLowerCase()
}

async function evaluateQuestion(question, answerOptions, correctAnswer, correctMessage, wrongMessage){
    // ask the user to answer the given question
    println(question);
    println(answerOptions.map((answer, index) => `${index+1})${answer}`).join(', '));
    // this is a blocking wait
    const answer = await input('?')
    const isCorrect = equalIgnoreCase(correctAnswer, answer)
    println(isCorrect ? correctMessage : wrongMessage)
    return isCorrect ? 1 : 0
}

async function main(){
    let score = 0

    printAlign("LITERATURE QUIZ", "center")
    printAlign("CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY", "center")
    println("\n\n")

    println("TEST YOUR KNOWLEDGE OF CHILDREN'S LITERATURE.");
    println();
    println("THIS IS A MULTIPLE-CHOICE QUIZ.");
    println("TYPE A 1, 2, 3, OR 4 AFTER THE QUESTION MARK.");
    println();
    println("GOOD LUCK!");
    println("\n\n");

    score += await evaluateQuestion("IN PINOCCHIO, WHAT WAS THE NAME OF THE CAT?",
        [ "TIGGER", "CICERO", "FIGARO", "GUIPETTO"], 3,
        "VERY GOOD!  HERE'S ANOTHER.", "SORRY...FIGARO WAS HIS NAME.")
    println()

    score += await evaluateQuestion("FROM WHOSE GARDEN DID BUGS BUNNY STEAL THE CARROTS?",
        [ "MR. NIXON'S", "ELMER FUDD'S", "CLEM JUDD'S", "STROMBOLI'S" ], 2,
        "PRETTY GOOD!", "TOO BAD...IT WAS ELMER FUDD'S GARDEN.")
    println()

    score += await evaluateQuestion("IN THE WIZARD OF OS, DOROTHY'S DOG WAS NAMED",
        [ "CICERO", "TRIXIA", "KING", "TOTO" ], 4,
        "YEA!  YOU'RE A REAL LITERATURE GIANT.",
        "BACK TO THE BOOKS,...TOTO WAS HIS NAME.")
    println()

    score += await evaluateQuestion("WHO WAS THE FAIR MAIDEN WHO ATE THE POISON APPLE",
        [ "SLEEPING BEAUTY", "CINDERELLA", "SNOW WHITE", "WENDY" ], 3,
        "GOOD MEMORY!", "OH, COME ON NOW...IT WAS SNOW WHITE.")

    println("\n")

    if(score === 4) {
        println("WOW!  THAT'S SUPER!  YOU REALLY KNOW YOUR NURSERY\n"+
        "YOUR NEXT QUIZ WILL BE ON 2ND CENTURY CHINESE\n"+
        "LITERATURE (HA, HA, HA)")
    } else if(score <= 2){
        println("UGH.  THAT WAS DEFINITELY NOT TOO SWIFT.  BACK TO\n" +
        "NURSERY SCHOOL FOR YOU, MY FRIEND.")
    } else {
        println("NOT BAD, BUT YOU MIGHT SPEND A LITTLE MORE TIME\n"+
        "READING THE NURSERY GREATS.")
    }
}

main()
